Feature/53 add user rud

jieun5119 opened the pull request:
resolved : #53 
## 📌 과제 설명
회원 정보 기능 구현 RUD
## 👩‍💻 요구 사항과 구현 내용 
-  마이페이지 로그인된 사용자 정보 조회
- 회원 정보 수정 (닉네임 / 핸드폰 번호)
- 비밀번호 수정
- 회원 탈퇴 기능 (softdelete)
- 회원 탈퇴 요청 후 30일이 지난 회원 정보가 삭제되는 스케줄링 구현
## ✅ 피드백 반영사항  
- 로그인 관련 피드백 반영
- 개행 수정
- 메소드 이름 변경
- 토큰 유지 시간 상수화
- User엔티티 Setter 제거
- 회원 정보 삭제 스케줄러 오류 수정
- 회원 수정 api 분리 
- 검사 메서드 분리 

## ✅ PR 포인트 & 궁금한 점
- 회원 정보 관련 RUD 부분 구현했습니다 . 부족한 부분은 피드백 주시면 감사하겠습니다

jieun5119 committed:
feat: 마이페이지 정보 조회

- 로그인된 회원의 정보 조회

jieun5119 committed:
feat: 마이페이지 정보 조회

- 로그인된 회원의 정보 조회

jieun5119 committed:
feat: 회원정보 수정 기능

- 로그인한 사용자의 닉네임, 핸드폰 번호 수정가능

jieun5119 committed:
feat: 회원정보 수정 기능

- 로그인한 사용자의 닉네임, 핸드폰 번호 수정가능

jieun5119 committed:
feat: 비밀번호 수정 기능 추가

jieun5119 committed:
feat: 비밀번호 수정 기능 추가

jieun5119 committed:
feat: 회원 탈퇴 기능 추가

- softdelete를 통한 회원 탈퇴기능

- 회원 탈퇴 요청 300일 후에 물리적 삭제하는 스케줄러 추가

jieun5119 committed:
feat: 회원 탈퇴 기능 추가

- softdelete를 통한 회원 탈퇴기능

- 회원 탈퇴 요청 300일 후에 물리적 삭제하는 스케줄러 추가

jieun5119 committed:
fix: 피드백 반영

- 메소드 이름 변경

- 토큰 유지시간 상수화

- 개행 수정

jieun5119 committed:
fix: 피드백 반영

- 메소드 이름 변경

- 토큰 유지시간 상수화

- 개행 수정

jhk01007 committed:
chore: coolsms 환경변수 추가 (#59)

l2yujw commented on src/main/java/com/sscanner/team/user/service/UserService.java:
```diff
@@ -61,8 +68,65 @@ public UserJoinResponseDto join(UserJoinRequestDto req){
+
+        if (phone != null && !user.getPhone().equals(phone)) {
+            checkDuplicatedPhone(phone);
+            user.setPhone(phone);
```
저도 피드백을 받았던 부분인데
setXXX 대신에 UserUpdateRequestDto에 toEntity를 만들어서 User값을 변경하는 것도 좋을거 같습니다!
그러면 req.nickname할 필요 없이 받아온거 그대로 toEntity에서 user만 새로 받아서 if문 넣을 수 있지 않을까 생각합니다!

l2yujw commented on src/main/java/com/sscanner/team/user/service/UserService.java:
```diff
@@ -61,8 +68,65 @@ public UserJoinResponseDto join(UserJoinRequestDto req){
+            throw new BadRequestException(ExceptionCode.PASSWORD_NOT_MATCH);
+        }
+
+        user.setPassword(passwordEncoder.encode(newPassword));
```
해당 부분도 위의 내용과 같은 생각입니다!

7zrv commented on src/main/java/com/sscanner/team/user/controller/UserController.java:
```diff
@@ -30,7 +34,36 @@ public ApiResponse<UserJoinResponseDto> registerUSer(@RequestBody @Valid UserJoi
+        String message = userService.changePassword(
+                requestDto.currentPassword(),
+                requestDto.newPassword(),
+                requestDto.confirmNewPassword()
```
파라미터로 requestDto 자체를 넣고 메서드 내부에서 dto 값들을 조작 해보는건 어떻게 생각하시나요?

7zrv commented on src/main/java/com/sscanner/team/user/repository/UserRepository.java:
```diff
@@ -3,13 +3,15 @@
     boolean existsByPhone(String phone);
     Optional<User> findByEmail(String email);
-
+    List<User> findAllByDeletedAt(LocalDateTime dateTime); // 회원 탈퇴 30일 지난 사용자
```
조회가 정상적으로 되시는지 궁금합니다! sqldelete 어노테이션이 조회를 막을거라는 생각이 들어서 여쭤봣습니다

7zrv commented on src/main/java/com/sscanner/team/user/service/ReissueService.java:
```diff
@@ -26,6 +26,9 @@ public class ReissueService {
 
     private static final String REFRESH_TOKEN = "refresh";
     private static final String ACCESS_TOKEN = "access";
+    private static final long ACCESS_TOKEN_EXPIRATION = 20000L; // 20초
```
토큰 지속시간을 20초로 하신이유를 여쭤보고싶습니다!

7zrv commented on src/main/java/com/sscanner/team/user/controller/UserController.java:
```diff
@@ -30,7 +34,36 @@ public ApiResponse<UserJoinResponseDto> registerUSer(@RequestBody @Valid UserJoi
+    }
 
+    //비밀번호 수정
+    @PostMapping("/change-password")
```
구현하신 업데이트는 Post보다 PatchMapping에 가까우신거 같습니다!

7zrv commented on src/main/java/com/sscanner/team/user/service/UserService.java:
```diff
@@ -61,8 +68,65 @@ public UserJoinResponseDto join(UserJoinRequestDto req){
+        String phone = req.phone();
+
+        if (nickname!= null && !user.getNickname().equals(nickname)) {
+            checkDuplicatedNickname(nickname);
```
중복 닉네임이 있는지를 먼저 검사하는게 좋지않을까 생각해봤습니다
아니면 닉네임 중복확인 메서드를 이용해서 프론트에서 중복검사를 하는 방식도 존재할 것 같습니다!

7zrv commented on src/main/java/com/sscanner/team/user/service/UserService.java:
```diff
@@ -61,8 +68,65 @@ public UserJoinResponseDto join(UserJoinRequestDto req){
+            throw new BadRequestException(ExceptionCode.CURRENT_PASSWORD_NOT_MATCH);
+        }
+
+        if (!newPassword.equals(newPasswordConfirm)) {
```
검사 메서드를 분리하는건 어떻게 생각하시나요?
제 생각엔 분리하고 문장처럼 정리하면 좀 더 직관적일것 같아서 말씀드렸습니다

7zrv commented on src/main/java/com/sscanner/team/User.java:
```diff
@@ -2,18 +2,16 @@
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
+@Setter
```
Setter 사용 대신 상태변화 메서드를 엔티티 내부에 두는게 어떠신가요?
캡슐화에 도움이 될거라고 생각했습니다!

l2yujw committed:
Feature/27 product img api (#47)

* refactor: 엔티티 디렉터리 이동

* refactor: 마지막 줄 빈칸 추가

* feat: S3 상품 이미지 등록 및 상품 별 이미지 ID 매칭 조회 API

* refactor: 개행 추가

* feat: 단일 상품에 대한 이미지들 조회 분리 및 전체 상품에 대한 이미지 조회 기능 추가

* feat: RedisDataInitializer and refactor: PointService 와 RedisManager 분리

* refactor: Points, Payment 디렉터리로 기능 분리

* feat: Redis에 변경된 User만 등록 and refactor: 코드 정리

* refactor: 스레드 풀 사용 / 변경으로 Redis에 등록된 것만 변경 / Cursor로 Redis 사용

* refactor: sonarcloud issues 적용

Leehunil committed:
Refactor/56 board comment api (#57)

* refactor : Board column 변경

- 관리자가 게시글의 승인 상태를 알 수 있도록 givePoint 컬럼에서 ApprovalStatus 컬럼으로 변경

* refactor : UserUtils 주입

- 게시글 생성 DTO에 userId 빼고 userUtils 사용
- BoardResponseDTO에 위치 정보 빼기

* refactor : board 삭제 로직 리펙토링

- 토큰 유저와 게시글 작성자가 일치하는지 확인하고 삭제를 진행한다.
- boardImg는 따로 삭제한다.

* refactor : board update 로직 리팩토링

- 토큰 유저와 게시글 작성자가 일치하는지 확인 후 업데이트 진행
- 업로드 파일이 비어있는지 확인하는 메서드를 따로 빼서 관리
- UserUtils 주입

* refactor : board 목록 조회 로직 리팩토링

- BoardListResponseDTO에 위치 정보 제외
- response 포멧 변경

* feat : board 위치 정보 로직 추가

- 지도 보여주기 버튼 누르면 게시글에 저장된 위치 정보 보여주기

* fix : 오류 변경

- 그 외 자잘한 오류 해결

* fix : 주석 제거

- 주석 제거

* refactor : Comment 생성 및 삭제 로직 refactoring

- @Valid 작성 누락해서 추가
- 댓글 삭제 시 토큰 유저와 댓글 작성자가 같은지 확인 과정 추가
- String은 NotNull 보다는 NotBlank 사용

* feat : Comments 조회 로직 추가

- CommentResponseDTO from으로 변경하여 Comment만 매개변수로 받아서 Comment.getUser()를 통해 유저 정보 전달

* feat : Board 삭제 시 모든 Comment도 삭제

- 벌크 연산을 사용하여 한 번의 update 쿼리로 논리 삭제를 하였습니다.

* fix : Transactional 누락

- Transactional 추가

jhk01007 committed:
Feature/48 add logging (#58)

* refactor: HealthController global/utils 패키지로 이동

* refactor: TimeTraceAspect 패키지 변경

* feat: log 관련 설정 파일 추가

- logback-spring.xml: 로그에 대한 전반적인 설정정보
- console-appender.xml: 콘솔에 나타날 로그 appender 정의.
- file-*-appender.xml: 파일로 저장될 로그 appender 정의.
- spring boot profile 기능을 통해서 개발환경에서는 콘솔에, 운영환경에서는 파일로 로그가 저장하도록 설정하였음.

* feat: 로그 레벨별로 글씨색깔 다르게 보이게하는 기능 추가

* refactor: 개발환경에서 로그 레벨 DEBUG로 변경

* refactor: 시간 측정 로그 메시지 영어로 변경

* feat: Controller layer 로그 기능 구현 완료

- AOP를 통해 controller 패키지 안에 있는 모든 클래스에 적용시켰습니다.
- MDC를 통해 멀티스레드 환경에서도 로그를 구분할 수 있게 하였습니다.
- 예외의 경우 GlobalExceptionHandler에서 정의한 예외는 비즈니스적인 예외라고 판단해 제외하고 나머지 예외에 대해서만 ERROR레벨의 로그를 적용시켰습니다.

* feat: Service 로그 기능 구현 완료

- 보통 서비스 로직은 컨트롤러를 통해서만 호출되기 때문에 MDC, 예외처리 로직을 추가하지 않았습니다.

* cicd: 배포 환경 프로필에 대한 환경변수 추가

* cicd: clean build에서 그냥 build로 수정

- 캐싱을 활용하기 위해서 수정하였습니다.

* cicd: Docker Compose 파일에 프로필관련 환경변수 추가

* test: 테스트를 위한 코드 추가

- 테스트를 위해 현재브랜치에 푸시할 시 파이프라인이 작동하게 함
- 임시코드이며 곧 삭제 예정

* test: 테스트를 위한 테스트용 컨트롤러, 서비스 추가

- 테스트 후 삭제 예정

* test: 테스트를 위해 SecurityConfig 임시 수정

- 테스트 후 롤백 예정

* refactor: 시간 측정 로그를 Http, Service AOP에 추가

- 동일 스레드로 처리하기 위해서 수정하였습니다.

* test: 테스트를 위해 사용했던 코드 삭제 및 원상 복구

* refactor: startTime에 static 키워드 제거

* chore: coolsms 환경변수 추가 (#59)

* refactor: HealthController global/utils 패키지로 이동

* refactor: TimeTraceAspect 패키지 변경

* feat: log 관련 설정 파일 추가

- logback-spring.xml: 로그에 대한 전반적인 설정정보
- console-appender.xml: 콘솔에 나타날 로그 appender 정의.
- file-*-appender.xml: 파일로 저장될 로그 appender 정의.
- spring boot profile 기능을 통해서 개발환경에서는 콘솔에, 운영환경에서는 파일로 로그가 저장하도록 설정하였음.

* feat: 로그 레벨별로 글씨색깔 다르게 보이게하는 기능 추가

* refactor: 개발환경에서 로그 레벨 DEBUG로 변경

* refactor: 시간 측정 로그 메시지 영어로 변경

* feat: Controller layer 로그 기능 구현 완료

- AOP를 통해 controller 패키지 안에 있는 모든 클래스에 적용시켰습니다.
- MDC를 통해 멀티스레드 환경에서도 로그를 구분할 수 있게 하였습니다.
- 예외의 경우 GlobalExceptionHandler에서 정의한 예외는 비즈니스적인 예외라고 판단해 제외하고 나머지 예외에 대해서만 ERROR레벨의 로그를 적용시켰습니다.

* feat: Service 로그 기능 구현 완료

- 보통 서비스 로직은 컨트롤러를 통해서만 호출되기 때문에 MDC, 예외처리 로직을 추가하지 않았습니다.

* cicd: 배포 환경 프로필에 대한 환경변수 추가

* cicd: clean build에서 그냥 build로 수정

- 캐싱을 활용하기 위해서 수정하였습니다.

* cicd: Docker Compose 파일에 프로필관련 환경변수 추가

* test: 테스트를 위한 코드 추가

- 테스트를 위해 현재브랜치에 푸시할 시 파이프라인이 작동하게 함
- 임시코드이며 곧 삭제 예정

* test: 테스트를 위한 테스트용 컨트롤러, 서비스 추가

- 테스트 후 삭제 예정

* test: 테스트를 위해 SecurityConfig 임시 수정

- 테스트 후 롤백 예정

* refactor: 시간 측정 로그를 Http, Service AOP에 추가

- 동일 스레드로 처리하기 위해서 수정하였습니다.

* test: 테스트를 위해 사용했던 코드 삭제 및 원상 복구

* refactor: startTime에 static 키워드 제거

* feat: /health 엔드포인트 permitAll()설정

- /health 엔드포인트의 경우 블루/그린 배포시 health check를 위해 열어놔야 합니다.

* test: 블루/그린 배포 시연

* test: 블루/그린 배포 시연코드 제거

* Feature/27 product img api (#47)

* refactor: 엔티티 디렉터리 이동

* refactor: 마지막 줄 빈칸 추가

* feat: S3 상품 이미지 등록 및 상품 별 이미지 ID 매칭 조회 API

* refactor: 개행 추가

* feat: 단일 상품에 대한 이미지들 조회 분리 및 전체 상품에 대한 이미지 조회 기능 추가

* feat: RedisDataInitializer and refactor: PointService 와 RedisManager 분리

* refactor: Points, Payment 디렉터리로 기능 분리

* feat: Redis에 변경된 User만 등록 and refactor: 코드 정리

* refactor: 스레드 풀 사용 / 변경으로 Redis에 등록된 것만 변경 / Cursor로 Redis 사용

* refactor: sonarcloud issues 적용

* Refactor/56 board comment api (#57)

* refactor : Board column 변경

- 관리자가 게시글의 승인 상태를 알 수 있도록 givePoint 컬럼에서 ApprovalStatus 컬럼으로 변경

* refactor : UserUtils 주입

- 게시글 생성 DTO에 userId 빼고 userUtils 사용
- BoardResponseDTO에 위치 정보 빼기

* refactor : board 삭제 로직 리펙토링

- 토큰 유저와 게시글 작성자가 일치하는지 확인하고 삭제를 진행한다.
- boardImg는 따로 삭제한다.

* refactor : board update 로직 리팩토링

- 토큰 유저와 게시글 작성자가 일치하는지 확인 후 업데이트 진행
- 업로드 파일이 비어있는지 확인하는 메서드를 따로 빼서 관리
- UserUtils 주입

* refactor : board 목록 조회 로직 리팩토링

- BoardListResponseDTO에 위치 정보 제외
- response 포멧 변경

* feat : board 위치 정보 로직 추가

- 지도 보여주기 버튼 누르면 게시글에 저장된 위치 정보 보여주기

* fix : 오류 변경

- 그 외 자잘한 오류 해결

* fix : 주석 제거

- 주석 제거

* refactor : Comment 생성 및 삭제 로직 refactoring

- @Valid 작성 누락해서 추가
- 댓글 삭제 시 토큰 유저와 댓글 작성자가 같은지 확인 과정 추가
- String은 NotNull 보다는 NotBlank 사용

* feat : Comments 조회 로직 추가

- CommentResponseDTO from으로 변경하여 Comment만 매개변수로 받아서 Comment.getUser()를 통해 유저 정보 전달

* feat : Board 삭제 시 모든 Comment도 삭제

- 벌크 연산을 사용하여 한 번의 update 쿼리로 논리 삭제를 하였습니다.

* fix : Transactional 누락

- Transactional 추가

* refactor: HealthController global/utils 패키지로 이동

* refactor: TimeTraceAspect 패키지 변경

* feat: log 관련 설정 파일 추가

- logback-spring.xml: 로그에 대한 전반적인 설정정보
- console-appender.xml: 콘솔에 나타날 로그 appender 정의.
- file-*-appender.xml: 파일로 저장될 로그 appender 정의.
- spring boot profile 기능을 통해서 개발환경에서는 콘솔에, 운영환경에서는 파일로 로그가 저장하도록 설정하였음.

* feat: 로그 레벨별로 글씨색깔 다르게 보이게하는 기능 추가

* refactor: 개발환경에서 로그 레벨 DEBUG로 변경

* refactor: 시간 측정 로그 메시지 영어로 변경

* feat: Controller layer 로그 기능 구현 완료

- AOP를 통해 controller 패키지 안에 있는 모든 클래스에 적용시켰습니다.
- MDC를 통해 멀티스레드 환경에서도 로그를 구분할 수 있게 하였습니다.
- 예외의 경우 GlobalExceptionHandler에서 정의한 예외는 비즈니스적인 예외라고 판단해 제외하고 나머지 예외에 대해서만 ERROR레벨의 로그를 적용시켰습니다.

* feat: Service 로그 기능 구현 완료

- 보통 서비스 로직은 컨트롤러를 통해서만 호출되기 때문에 MDC, 예외처리 로직을 추가하지 않았습니다.

* cicd: 배포 환경 프로필에 대한 환경변수 추가

* cicd: clean build에서 그냥 build로 수정

- 캐싱을 활용하기 위해서 수정하였습니다.

* cicd: Docker Compose 파일에 프로필관련 환경변수 추가

* test: 테스트를 위한 코드 추가

- 테스트를 위해 현재브랜치에 푸시할 시 파이프라인이 작동하게 함
- 임시코드이며 곧 삭제 예정

* test: 테스트를 위한 테스트용 컨트롤러, 서비스 추가

- 테스트 후 삭제 예정

* test: 테스트를 위해 SecurityConfig 임시 수정

- 테스트 후 롤백 예정

* refactor: 시간 측정 로그를 Http, Service AOP에 추가

- 동일 스레드로 처리하기 위해서 수정하였습니다.

* test: 테스트를 위해 사용했던 코드 삭제 및 원상 복구

* refactor: startTime에 static 키워드 제거

* feat: Service 로그 기능 구현 완료

- 보통 서비스 로직은 컨트롤러를 통해서만 호출되기 때문에 MDC, 예외처리 로직을 추가하지 않았습니다.

* test: 테스트를 위한 코드 추가

- 테스트를 위해 현재브랜치에 푸시할 시 파이프라인이 작동하게 함
- 임시코드이며 곧 삭제 예정

* test: 테스트를 위한 테스트용 컨트롤러, 서비스 추가

- 테스트 후 삭제 예정

* test: 테스트를 위해 SecurityConfig 임시 수정

- 테스트 후 롤백 예정

* refactor: 시간 측정 로그를 Http, Service AOP에 추가

- 동일 스레드로 처리하기 위해서 수정하였습니다.

* test: 테스트를 위해 사용했던 코드 삭제 및 원상 복구

* feat: /health 엔드포인트 permitAll()설정

- /health 엔드포인트의 경우 블루/그린 배포시 health check를 위해 열어놔야 합니다.

* test: 블루/그린 배포 시연

* test: 블루/그린 배포 시연코드 제거

---------

Co-authored-by: l2yuPa <49338509+l2yujw@users.noreply.github.com>
Co-authored-by: Lee hunil <104710245+Leehunil@users.noreply.github.com>

jieun5119 committed:
fix: 회원정보 삭제 스케줄러 오류수정

jieun5119 committed:
fix: 회원정보 삭제 스케줄러 오류수정

jieun5119 committed:
fix: 상태변화 메서드 추가

 - Setter제거 후 상태변화 메서드 추가

jieun5119 committed:
fix: 상태변화 메서드 추가

 - Setter제거 후 상태변화 메서드 추가

jieun5119 committed:
fix: 가입시 권한부여 수정

jieun5119 committed:
fix: 가입시 권한부여 수정

jieun5119 committed:
fix : 로그인 전송 방식 수정

 - logindto 추가

 - 로그인 전송방식 폼데이터에서 JSON으로 변경

jieun5119 committed:
fix : 로그인 전송 방식 수정

 - logindto 추가

 - 로그인 전송방식 폼데이터에서 JSON으로 변경

jieun5119 committed:
fix: 필요없는 주석 제거

jieun5119 committed:
fix: 필요없는 주석 제거

jieun5119 committed:
fix : 회원 정보 수정 api 분리

- 닉네임 변경 , 폰번호 변경 api 분리

jieun5119 committed:
fix : 회원 정보 수정 api 분리

- 닉네임 변경 , 폰번호 변경 api 분리

jieun5119 committed:
fix : 패키지명 변경

- 패키지명 변경 (sonar 이슈)

jieun5119 committed:
fix : 패키지명 변경

- 패키지명 변경 (sonar 이슈)

jieun5119 committed:
fix : 회원탈퇴 api 반환타입 수정

jieun5119 committed:
fix : 회원탈퇴 api 반환타입 수정

l2yujw committed:
Feature/61 add gifticon (#62)

* feat: Barcode 생성 서비스 (zxing)

* feat: 기프티콘 생성 서비스 및 조회 API

* feat: 결제시 바코드 생성 및 결제 내역에 url 추가

7zrv commented on src/main/java/com/sscanner/team/User.java:
```diff
@@ -2,18 +2,16 @@
+@Where(clause = "deleted_at is NULL")
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
 @Table(name = "USER")
 public class User extends BaseEntity {
```
유저 엔티티를 user 디렉토리에 entity 디렉토리 만들어서 넣어주시면 좋을것 같아요!

jieun5119 replied:
넵 알겠습니다 ! 👍

jhk01007 committed:
feat: ELK 환경변수 추가 (#65)

jieun5119 committed:
fix: User엔티티 디렉토리 이동

jieun5119 committed:
fix: User엔티티 디렉토리 이동

jieun5119 committed:
Revert "fix : 회원탈퇴 api 반환타입 수정"

This reverts commit fcc17febf343752473ef645073dbbce6d0b5ab85.

jieun5119 committed:
Merge remote-tracking branch 'origin/feature/53-add-user-RUD' into feature/53-add-user-RUD

# Conflicts:
#	src/main/java/com/sscanner/team/payment/entity/PaymentRecord.java
#	src/main/java/com/sscanner/team/payment/requestdto/PaymentRequestDto.java
#	src/main/java/com/sscanner/team/points/service/PaymentServiceImpl.java
#	src/main/java/com/sscanner/team/user/controller/UserController.java

jieun5119 committed:
Revert "fix : 회원탈퇴 api 반환타입 수정"

This reverts commit fcc17feb

jieun5119 committed:
"fix: 디렉토리 변경에 따른 import수정"

jieun5119 committed:
fix : 리베이스 후 불필요한 부분 삭제
